ocs_ci/ocs/node.py

```diff
@@ -771,13 +772,30 @@ def get_ocs_nodes(num_of_nodes=None):
             it returns all the ocs nodes.
 
     Returns:
-        list: List of ocs nodes
+        list: List of the ocs nodes
 
     """
-    ocs_node_names = machine.get_labeled_nodes(constants.OPERATOR_NODE_LABEL)
+    # Import inside the function to avoid circular loop
+    from ocs_ci.ocs.cluster import is_managed_service_cluster
+    from ocs_ci.ocs.resources.storage_cluster import get_storage_cluster_state
+
+    # If we use managed service or external mode the worker nodes are without the OCS label.
+    # So in that case, we will get the worker nodes without searching for the OCS label.
+    ms_with_odf_addon = is_managed_service_cluster() and is_odf_addon_installed()
```

Comment:
And if the addon is being installed and it's not yet in ready state?

Reply:
IIUC, It does mean that it's not correctly installed. I took the idea from here: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/rosa.py#L302.